pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/SetOperator.java

```diff
@@ -112,15 +112,14 @@ protected void constructRightBlockSet() {
 
   protected TransferableBlock constructResultBlockSet(TransferableBlock leftBlock) {
     List<Object[]> rows = new ArrayList<>();
+    // TODO: Other operators keep the first erroneous block, while this keep the last.
+    //  We should decide what is what we want to do and be consistent with that.
```

Comment:
I'm not 100% sure about this, but we should decide what to do in these cases

Reply:
I think once we read an error from an operator, we don't read that operator again. But in case we do, what should we return? The first error or a new one? Right now some operators keep the first error emitted and always return the same error afterwards, but others create a new error each time they are being called.

My TODO is to try to homogenize that to either always return the first error or always create a new one. Alternative, we can define that we cannot call `getNextBlock()` (or read from mailboxes) once the first error is emitted. In that case we should throw an exception.